charts/scalardb-cluster/templates/scalardb-cluster/role.yaml

```diff
@@ -7,3 +7,8 @@ rules:
   - apiGroups: [""]
     resources: ["endpoints"]
     verbs: ["get", "watch", "list"]
+  {{- if and (.Values.scalardbCluster.tde.enabled) (eq .Values.scalardbCluster.tde.type "self") }}
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["get", "create", "delete"]
+  {{- end }}
```

Comment:
If we use the `self` type TDE, we must allow ScalarDB Cluster's pods to create, get, and delete the Kubernetes Secrets resources. This is because ScalarDB Cluster manages the data encryption keys by using the Kubernetes Secrets.